src/models/dapplets.ts

```diff
@@ -197,105 +227,107 @@ const reducers = {
 };
 
 const effects = (dispatch: any) => ({
-  //
-  getDapplets: async (): Promise<void> => {
+  getDapplets: async (payload: number, rootState: any): Promise<void> => {
     const provider = new ethers.providers.JsonRpcProvider(PROVIDER_URL, 0x05);
-    const contract: any = new ethers.Contract(
-      "registry.dapplet-base.eth",
-      abi,
+
+    const dappletsRegistry = new ethers.Contract(
+      DAPPLET_REGISTRY_ADDRESS,
+      dappletsRegistryABI,
       provider,
     );
-    const events = await contract.queryFilter("ModuleInfoAdded");
-
-    async function getVersions(name: string) {
-      const hex: string = await contract.getVersionNumbers(name, "default");
-      const result = (hex.replace("0x", "").match(/.{1,8}/g) ?? []).map(
-        (x) =>
-          `${parseInt("0x" + x[0] + x[1])}.${parseInt(
-            "0x" + x[2] + x[3],
-          )}.${parseInt("0x" + x[4] + x[5])}`,
-      );
-      return result;
-    }
-    const dapplets: DappletsList = {};
-    const myPromises = [];
-    for (let i = 1; i <= events.length; i++) {
-      myPromises.push(
-        contract.modules(i).then(async (module: any) => {
-          if (module.moduleType !== 1) return;
-          const ev = events[i - 1];
-          const block = await ev.getBlock();
-          const versions = await getVersions(module.name);
-          const date = new Date(block.timestamp * 1000);
-          const pad = (n: number, s = 2) =>
-            `${new Array(s).fill(0)}${n}`.slice(-s);
-          const icon = {
-            hash: module.icon.hash,
-            uris: module.icon.uris.map((u: any) =>
-              ethers.utils.toUtf8String(u),
-            ),
-          };
-          // const url: string = await _getResource(icon)
-          const dapplet: IDapplet = {
-            id: i,
-            description: module.description,
-            icon: "",
-            name: module.name,
-            owner: module.owner,
-            title: module.title,
-            versionToShow: versions[versions.length - 1],
-            version: versions,
-            timestampToShow: `${pad(date.getDay())}.${pad(
-              date.getMonth() + 1,
-            )}.${pad(date.getFullYear(), 4)}`,
-            timestamp: block.timestamp,
-            trustedUsers: [module.owner],
-            isExpanded: false,
-            interfaces: [],
-          };
-          await dispatch.dapplets.addDapplet(dapplet);
-          _getResource(icon).then((url) => {
-            dispatch.dapplets.setBlobUrl({ id: i, blobUrl: url });
-          });
-          dapplets[`${dapplet.id}`] = dapplet;
-        }),
-      );
-    }
-    await Promise.all(myPromises);
-    const contractListing: any = new ethers.Contract(
-      "0xD8298EBF33a44e954AeaC125231547A1426Cb412",
-      abiListing2,
-      provider,
+
+    const data = await dappletsRegistry.getModules(0, MAX_MODULES_COUNTER);
```

Comment:
You mean that the max value for the secon arg is 20? no matter what number I pass, it wont matter if it's over 20?

Reply:
You are able to use any `limit` size (page size). There is a risk of out of gas when use too big values. 
If the registry contains more modules than `limit` amount, then you have to iterate by pages changing the `offset` parameter. `await dappletsRegistry.getModules(offset, limit);` otherwise, not all modules will be displayed in the store